### PR TITLE
fix: add image fallback

### PR DESCRIPTION
**Pull request by KrissDrawing**
# Pull Request Template

[#3436](https://github.com/bloom-housing/bloom/issues/3436)

## Description

PR adds ability to display a fallback image on loading error. 
If i get this correctly regular `` tags doesn't call `onError` when rendered on server. So for this to work i added lazy loading. Really can't tell if that will negatively affect website, it doesn't seem like, but it's something to have in mind.

## How Can This Be Tested/Reviewed?

Provide instructions so we can review.

To test this i manually typed invalid urls, so either here type a string in `src` or in `ListingView` and `helpers` in bloom repo. 
To test:
- when going from site to site
- when refreshing the page
- /listing page
- /listing/[id] page
- modal images in multiple photos

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have made any corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added or updated stories if new changes are not captured in Storybook
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have exported any new pieces added to ui-components
- [ ] I have documented this change in the changelog, and any breaking changes are well described

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Either review the Storybook preview or pull the changes down locally and test that the acceptance criteria is met
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

On merge, squash commits.

### CODE REVIEW

**Review by jaredcwhite — COMMENTED** [review]
@KrissDrawing I like this idea a lot! One thought I have though re: lazy loading images. That makes sense to me if you're on the Listings view and there are tons of listings…no need to download a ton of images everywhere until you scroll down far enough to see them. However, for the individual Listing view & when the gallery modal comes up, I think you'd want all those images to display as quickly as possible. So I'm not sure if those should be lazy loading. Also FYI, I don't see any Listing**s** view update along these lines in PR https://github.com/bloom-housing/bloom/pull/3551

**Review by ColinBuyck — COMMENTED** [review]
This is an interesting one! My gut reaction is hesitation to make loading adjustments in order to support the fallback but see your points... I'm wondering if you've looked at either of these sources https://stackoverflow.com/questions/8124866/how-does-one-use-the-onerror-attribute-of-an-img-element or https://github.com/vercel/next.js/discussions/14772. If these don't help much, do you mind providing some resources to support this approach?

**Comment by KrissDrawing** [comment]
I believe that all of mentioned solutions force it to  handle "onError" on client side, so lazy loading images, loading `ImageCard` using `next/dynamic` with `ssr: false` option, having image url in useState and set it in useEffect. 
But i found that hook: https://github.com/facebook/react/issues/15446#issuecomment-484709466
and in terms of loading valid images it seem to behave better than previous solution in fact similar to the one we got now. if `onError` triggers after hydration it will create layout shift (so for a while default "error image" state will be visible) but that's only when image is invalid (so it kind of works as on client side). I can see in that topic there are other solutions with `dangerouslySetInnerHTML`, but i don't really feel to good about those 😄 
If you strongly against that solution let me know @ColinBuyck @jaredcwhite but tbh i don't see anything better for now 😞

**Comment by jaredcwhite** [comment]
Maybe I'm missing a beat here, but do we have a big problem of image URLs coming back as 404 Not Found or whatever? Shouldn't we be doing a better job of enforcing valid image URLs on the server side first?

File: src/helpers/useFallbackImage.ts
Comment (ludtkemorgan) [inline src/helpers/useFallbackImage.ts:17]: Is there a way you have been testing with more than one image in an ImageCard?

**Review by ludtkemorgan — COMMENTED** [review]
I think this technique seems good. It works on the public site and I like the use of the custom hook. I'll hold off on approving this until the tests are implemented

File: src/helpers/useFallbackImage.ts
Comment (KrissDrawing) [reply src/helpers/useFallbackImage.ts:19]: i tested it for example in `Test: Coliseum` listing by replacing images prop with sth like this (`listingView.tsx` line 511):
```
images={imageUrlFromListing(listing, parseInt(process.env.listingPhotoSize)).map(
            (imageUrl: string, index) => {
              return {
                url: index % 2 ? imageUrl : "test",
              }
            }
          )}
```

**Review by ludtkemorgan — APPROVED** [review]

**Review by emilyjablonski — APPROVED** [review]
This LGTM, if it's easy would love to see a story for this variant, if not no worries

**Pull request merged**
